core/trino-spi/src/main/java/io/trino/spi/type/RowType.java

```diff
@@ -134,6 +136,15 @@ private RowType(TypeSignature typeSignature, List<Field> originalFields)
 
         this.comparable = fields.stream().allMatch(field -> field.getType().isComparable());
         this.orderable = fields.stream().allMatch(field -> field.getType().isOrderable());
+
+        // flat fixed size is one null byte for each field plus the sum field fixed sizes
+        int fixedSize = fieldTypes.size();
+        for (Type fieldType : fieldTypes) {
+            fixedSize += fieldType.getFlatFixedSize() + 4;
```

Comment:
Where does the `4` come from? "one null byte" should be 1, no?

Reply:
Same question, though the calculation is copied from below